matchers.go

```diff
@@ -1564,3 +1569,19 @@ func HaveSuccessfulRollout() gtypes.GomegaMatcher {
 	rm := &RolloutMatcher{}
 	return rm
 }
+
+func GVK(
+	gvk schema.GroupVersionKind,
+	optionalClient ...client.Client,
+) func() (*apiextensionsv1.CustomResourceDefinition, error) {
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	c := handleDefaultClient(optionalClient...)
+	key := util.GVKToObjectKey(gvk)
+	return func() (*apiextensionsv1.CustomResourceDefinition, error) {
+		err := c.Get(context.Background(), key, crd)
+		if apierrors.IsNotFound(err) {
+			err = nil
+		}
+		return crd, err
```

Comment:
Ah actually this will fail if users do not 
```
apiextensionsv1.AddToScheme(client)
```
beforehand... perhaps a magic sync.Once that registers this to the scheme when we call `GVK`? WDYT?